osu.Game/Rulesets/Objects/Legacy/ConvertHitObjectParser.cs

```diff
@@ -523,25 +477,61 @@ public int CustomSampleBank
             /// Layered hit samples are automatically added in all modes (except osu!mania), but can be disabled
             /// using the <see cref="LegacySkinConfiguration.LegacySetting.LayeredHitSounds"/> skin config option.
             /// </remarks>
-            public bool IsLayered { get; set; }
+            public readonly bool IsLayered;
+
+            public LegacyHitSampleInfo(string name, string? bank = null, int volume = 0, int customSampleBank = 0, bool isLayered = false)
+                : base(name, bank, customSampleBank >= 2 ? customSampleBank.ToString() : null, volume)
+            {
+                CustomSampleBank = customSampleBank;
+                IsLayered = isLayered;
+            }
+
+            public override HitSampleInfo With(Optional<string> name = default, Optional<string?> bank = default, Optional<string?> suffix = default, Optional<int> volume = default)
+                => With(name, bank, volume);
```

Comment:
This is probably the shadiest corner of this PR, what with `suffix` getting outright dropped if somebody uses this override by accident - not seeing a good way to avoid this, though... I'd probably outright throw and advise not to use this instead.

Reply:
I'm fine with throwing if suffix is non-default in this case, however what should I do about `FileHitSampleInfo`? It discards all except volume, and it's going to hit all of those via `SampleControlPoint.ApplyTo(sample)` that DHOs call.